libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -774,6 +774,94 @@ void ArrayUtils::resizeDynamicArray(ArrayType const& _typeIn) const
 	);
 }
 
+void ArrayUtils::popStorageArrayElement(ArrayType const& _type) const
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	if (!_type.isByteArray() && _type.baseType()->storageBytes() < 32)
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+
+	if (_type.isByteArray())
+	{
+		m_context.appendInlineAssembly(R"({
+			let slot_value := sload(ref)
+			switch and(slot_value, 1)
+			case 0 {
+				// short byte array
+				let length := and(div(slot_value, 2), 0x3f)
+				if iszero(length) { invalid() }
+
+				// Zero-out the suffix of the byte array by masking it.
+				// Do not zero-out the least significant byte, but mask the
+				// higher bits of the length.
+				// (((1<<(8 * (32 - length))) - 1) << 8) + 128
+				let mask := add(mul(0x100, sub(exp(0x100, sub(32, length)), 1)), 0xc0)
```

Comment:
It probably would make sense somehow using the `shiftLeftFunction` from `ABIFunctions` here or better, why not create the mask in C++?